### ISSUE TITLE
Panic when using lazy on a repo with ~25k files

### ISSUE THREAD

**Issue by rj3d** [pachyderm/pachyderm#2006]
I'm trying to use lazy loading loading on a repo with ~25k files. @sjezewski informs me that it's not possible, but instead of failing the containers panic and the pipeline looks as if it's running when inspecting with `pachctl list-job`. I've attached the full output of kubectl logs for user and storage for one of the pods.

[Terminal Saved Output.txt](https://github.com/pachyderm/pachyderm/files/1116463/Terminal.Saved.Output.txt)

**Comment by derekchiang** [comment]
Yeah we've seen that happen before.  Basically your containers can fail due to the underlying OS running out of threads, or if you hit the Go runtime thread limit (which I think is 10k), since we currently spawn one OS thread per lazy file (which is admittedly very much less than ideal).  It's on our roadmap to address this issue soon.

**Comment by sjezewski** [comment]
This is still a bug, we should report this better

**Comment by DSchmidtDev** [comment]
@derekchiang It seems that I'm also hitting the Go runtime thread limit (all Subconns are in TransientFailure) when processing 10k-25k files but I explicitly set `lazy` to `false`. 

Is it possible that the file handling is independent of the lazy option?
